api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java

```diff
@@ -64,45 +65,50 @@
     public ResponseEntity<SubjectListResponse> searchSubjects(SearchRequest request) {
         /* TODO: this function currently processes a SearchGroupItem; higher level handling will be needed */
         SearchGroupItem item = request.getInclude().get(0).get(0);
-        QueryRequest query;
-        QueryResult result;
-        Map<String, Integer> resultMapper;
-
         SubjectListResponse subjectSet = new SubjectListResponse();
 
-        if (item.getType().equals("ICD9")) {
-            List<SearchParameter> params = item.getSearchParameters();
-
-            /*  Check for SearchParameters without Domain IDs.
-                If any are present, generate a query to retrieve the missing domain IDs, then append the new parameters
-                to the parameter set
-             */
-            List<String> paramsWithoutDomains = generator.findParametersWithEmptyDomainIds(params);
-            if (!paramsWithoutDomains.isEmpty()) {
-                query = generator.findGroupCodes(item.getType(), paramsWithoutDomains);
-                result = executeQuery(query);
-                Map <String, Integer> resultMap = getResultMapper(result);
-                for (List<FieldValue> row: result.iterateAll()) {
-                    final SearchParameter newParam = new SearchParameter();
-                    newParam.setDomainId(row.get(resultMap.get("domainId")).getStringValue());
-                    newParam.setCode(row.get(resultMap.get("code")).getStringValue());
-                    params.add(newParam);
-                }
-            }
+        List<SearchParameter> paramsWithDomains = filterSearchParametersWithDomain(item.getSearchParameters());
+        List<SearchParameter> paramsWithoutDomains = filterSearchParametersWithoutDomain(item.getSearchParameters());
+
+        /** TODO: this is temporary and will be removed when we figure out the conceptId mappings **/
+        if (!paramsWithoutDomains.isEmpty()) {
+            QueryRequest queryRequest = QueryBuilderFactory
+                    .getQueryBuilder(FactoryKey.getKey("GROUP_CODES"))
```

Comment:
Can't you just use FactoryKey.GROUP_CODES?

Reply:
Yep.